CONTRIBUTING.md

```diff
@@ -1,3 +1,189 @@
+
 Welcome! As a [Jupyter](https://jupyter.org) project, we follow the [Jupyter contributor guide](https://jupyter.readthedocs.io/en/latest/contributor/content-contributor.html).
+
+## Types of contribution
+
+There are many ways to contribute to kubespawner, here are some of them:
+
+* **Update the documentation.**
+  If you're reading a page or docstring and it doesn't make sense (or doesn't exist!), please let us know by opening a bug report.
+  It's even more amazing if you can give us a suggested change.
+* **Fix bugs or add requested features.**
+  Have a look through the [issue tracker](https://github.com/jupyterhub/kubespawner/issues) and see if there are any tagged as ["help wanted"](https://github.com/jupyterhub/kubespawner/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22).
+  As the label suggests, we'd love your help!
+* **Report a bug.**
+  If kubespawner isn't doing what you thought it would do then open a [bug report](https://github.com/jupyterhub/kubespawner/issues/new).
+  Please provide details on what you were trying to do, what goal you were trying to achieve and how we can reproduce the problem.
+* **Suggest a new feature.**
+  We know that there are lots of ways to extend kubespawner!
+  If you're interested in adding a feature then please open a [feature request](https://github.com/jupyterhub/kubespawner/issues/new?template=feature_request.md).
+  Try to explain what the feature is, what alternatives you have though about, what skills are required to work on this task and how big a task you estimate it to be.
+* **Review someone's Pull Request.**
+  Whenever somebody proposes changes to the kubespawner codebase, the community reviews
+  the changes, and provides feedback, edits, and suggestions. Check out the
+  [open pull requests](https://github.com/jupyterhub/kubespawner/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc)
+  and provide feedback that helps improve the PR and get it merged. Please keep your
+  feedback positive and constructive!
+* **Tell people about kubespawner.**
+  Kubespawner is built by and for its community.
+  If you know anyone who would like to use kubespawner, please tell them about the project!
+  You could give a talk about it or run a demonstration or make a poster.
+  The sky is the limit :rocket::star2:.
+
+
+## Setting up for documentation changes
+
+We use Sphinx to build the kubespawner documentation. You can make changes to
+the documentation with any text editor and directly through the GitHub website.
+
+For small changes (like typos) you do not need to setup anything locally. For
+larger changes we recommend you build the documentation locally so you can see
+the end product in its full glory.
+
+To make edits through the GitHub website visit https://github.com/jupyterhub/kubespawner/tree/master/docs/source, open the file you would like to edit and then click "edit". GitHub will
+walk you through the process of proposing your change ("making a Pull Request").
+
```

Comment:
everything above here is "new" in the sense that it didn't use to be anywhere in the kubespawner repo. it is "old" in the sense that it is copied more or less directly from repo2docker's contrib guide.